torch/csrc/jit/script/python_sugared_value.cpp

```diff
@@ -660,6 +660,12 @@ std::shared_ptr<SugaredValue> toSugaredValue(
     if (auto callee = as_function(compiled_fn)) {
       return std::make_shared<FunctionValue>(*callee);
     }
+  }
+
+  py::bool_ isMethod = py::module::import("inspect").attr("ismethod")(obj);
+  // methods here have been explicitly annotated to not be compiled,
+  // so they do not have the same overload and compile checks as for functions
+  if (isFunction || isMethod) {
```

Comment:
You can also do `if (PyCallable_Check(obj.ptr()))` and avoid the extra `inspect` call

Reply:
Wouldn't that give a false positive for something that has a `__call__` function ?